packages/prisma-client/prisma/migrations/20221218211129_tree_text/migration.ts

```diff
@@ -0,0 +1,64 @@
+import { PrismaClient } from "./client";
+
+type PrevInstance = {
+  id: string;
+  component: any;
+  cssRules: Array<any>;
+  children: Array<PrevInstance | string>;
+};
+
+type Text = {
+  type: "text";
+  value: string;
+};
+
+type Instance = {
+  type: "instance";
+  id: string;
+  component: any;
+  cssRules: Array<any>;
+  children: Array<Instance | Text>;
+};
+
+const convertTree = (prevInstance: PrevInstance): Instance => {
+  const instance: Instance = {
+    type: "instance",
+    id: prevInstance.id,
+    component: prevInstance.component,
+    cssRules: prevInstance.cssRules,
+    children: [],
+  };
+
+  for (const child of prevInstance.children) {
+    if (typeof child === "string") {
+      instance.children.push({ type: "text", value: child });
+    } else {
+      instance.children.push(convertTree(child));
+    }
+  }
+  return instance;
+};
+
+export default () => {
+  const client = new PrismaClient({
+    // Uncomment to see the queries in console as the migration runs
+    // log: ["query", "info", "warn", "error"],
+  });
+  return client.$transaction(
+    async (prisma) => {
+      const previousTrees = await prisma.tree.findMany();
+      const update = previousTrees.map((tree) => {
+        return {
+          id: tree.id,
+          root: JSON.stringify(convertTree(JSON.parse(tree.root))),
```

Comment:
I'm not 100% sure all trees in prod DB are valid / in the current format. Might be a good idea to parse with zod. But maybe I'm overthinking.

Reply:
Wrapped with try/catch and found a few cases